crates/matrix-sdk-indexeddb/src/crypto_store/mod.rs

```diff
@@ -317,6 +364,197 @@ impl IndexeddbCryptoStore {
+                key: JsValue::from_str(keys::BACKUP_KEY_V1),
+                value: self.serializer.serialize_value(&a)?,
+            });
+        }
+
+        if !changes.sessions.is_empty() {
+            let sessions = indexeddb_changes.get(keys::SESSION);
+
+            for session in &changes.sessions {
+                let sender_key = session.sender_key().to_base64();
+                let session_id = session.session_id();
+
+                let pickle = session.pickle().await;
+                let key = self.serializer.encode_key(keys::SESSION, (&sender_key, session_id));
+
+                sessions.push(DbOperation::PutKeyVal {
+                    key,
+                    value: self.serializer.serialize_value(&pickle)?,
+                });
+            }
+        }
+
+        if !changes.inbound_group_sessions.is_empty() {
+            let sessions = indexeddb_changes.get(keys::INBOUND_GROUP_SESSIONS_V2);
+
+            for session in &changes.inbound_group_sessions {
+                let room_id = session.room_id();
+                let session_id = session.session_id();
+                let key = self
+                    .serializer
+                    .encode_key(keys::INBOUND_GROUP_SESSIONS_V2, (room_id, session_id));
+                let value = self.serialize_inbound_group_session(session).await?;
+                sessions.push(DbOperation::PutKeyVal { key, value });
+            }
+        }
+
+        if !changes.outbound_group_sessions.is_empty() {
+            let sessions = indexeddb_changes.get(keys::OUTBOUND_GROUP_SESSIONS);
+
+            for session in &changes.outbound_group_sessions {
+                let room_id = session.room_id();
+                let pickle = session.pickle().await;
+                sessions.push(DbOperation::PutKeyVal {
+                    key: self.serializer.encode_key(keys::OUTBOUND_GROUP_SESSIONS, room_id),
+                    value: self.serializer.serialize_value(&pickle)?,
+                });
+            }
+        }
+
+        let device_changes = &changes.devices;
+        let identity_changes = &changes.identities;
+        let olm_hashes = &changes.message_hashes;
+        let key_requests = &changes.key_requests;
+        let withheld_session_info = &changes.withheld_session_info;
+        let room_settings_changes = &changes.room_settings;
+
+        let device_store = indexeddb_changes.get(keys::DEVICES);
+
+        if !device_changes.new.is_empty() || !device_changes.changed.is_empty() {
+            for device in device_changes.new.iter().chain(&device_changes.changed) {
+                let key = self
+                    .serializer
+                    .encode_key(keys::DEVICES, (device.user_id(), device.device_id()));
+                let device = self.serializer.serialize_value(&device)?;
+
+                device_store.push(DbOperation::PutKeyVal { key, value: device });
+            }
+        }
+
+        if !device_changes.deleted.is_empty() {
```

Comment:
nit: this condition isn't useful, as the loop below won't iterate at all if it's empty